### PR TITLE
Role Contract Address as Env Var

### LINKED ISSUE #26: Roles contract doesn't exist on mainnet
This contract only exists on Sepolia.

https://github.com/cowdao-grants/mb-billing/blob/f997e6dafbe41b3f56559e405df6b6f0e2ec36b9/src/billingContract.ts#L36

@cowanator did you deploy this contract (ie. will it be different for every Safe) or is this a singleton contract that only exists once per chain and was deployed by the Zodiac team?

If the former, it should probably become an environment variable.

### PR DESCRIPTION
**Pull request by cowanator**
We had previously hard-coded the role contract. This now extracts from environment. Note that there were a few more changes than expected because we had a few places where we were checking for existence of role related fields. To simplify these checks we decided that role data must be provided and that we will always execute with role.

However, this comes with the slight inconvenience of having to provide role data for billing (although we do not execute billing via roles). A "workaround" is to provide empty role key and some invalid role contract address for billing (the code path will not hit any expectations of role data).

Closes #26 

### Test Plan

Existing CI

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -2,9 +2,11 @@
 ## Testnet
 RPC_URL=https://rpc.ankr.com/eth_sepolia
 BILLING_CONTRACT_ADDRESS=0xF1436859a0F04A827b79F8c92736F6331ebB64A1
+ROLE_CONTRACT_ADDRESS=0xa2f93c12E697ABC34770CFAB2def5532043E26e9
```
Comment (fleupold) [inline .env.sample:5]: naming nit:

ZODIAC_ROLES_MOD
ZODIAC_ROLE_KEY

**Review by fleupold — APPROVED** [review]
Looks good. I would like to make the dependency on Zodiac a bit more explicit in the naming.

File: .env.sample (hunk 1 above)
Comment (cowanator) [reply .env.sample]: 4d20077

**Comment by cowanator** [comment]
All updated and ready for merge!

**Pull request merged**